### ISSUE TITLE
Default anchor documentation incorrect for horizontal text (`text` and `textbbox`)

### ISSUE THREAD

**Issue by djhoese** [python-pillow/Pillow#7586]
### What did you do?

Read the documentation and the source code. I was mostly updating code to use `textbbox` instead of `textsize` and couldn't figure out why my text wasn't lining up the way I expected. The documentation for `text` and `textbbox` both say `anchor` defaults to "top left". This is only true for vertical (non-horizontal/) text.

https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html#PIL.ImageDraw.ImageDraw.text

https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html#PIL.ImageDraw.ImageDraw.textbbox

### What did you expect to happen?

I expected similar results between the old `textsize` and the newer `textbbox` so I could adjust the position of the text for the old code I was updating. This old code adjusts the position rather than adjust the anchor to support multiple drawing backends (avoid passing `anchor` which other drawing backends might not understand).

### What actually happened?

Unless I'm misreading the source code, the default anchor for horizontal text is "la":

https://github.com/python-pillow/Pillow/blob/54c5631fe1cbe37f9ba98c5bfedf790cb115347f/src/_imagingft.c#L624-L626

### What are your OS, Python and Pillow versions?

* OS: Ubuntu
* Python: 3.11
* Pillow: 10.1.0

```python
from PIL import Image, ImageDraw, ImageFont

font = ImageFont.load_default(size=40)
img = Image.new("RGB", (800, 800))
draw = ImageDraw.Draw(img)

text = "A"
pos = (400, 400)
bbox = draw.textbbox(pos, text, font, anchor="lt")
print("'lt' anchor bbox: ", bbox)
bbox = draw.textbbox(pos, text, font, anchor="la")
print("'la' anchor bbox: ", bbox)
bbox = draw.textbbox(pos, text, font)
print("Default anchor bbox: ", bbox)
```

```
'lt' anchor bbox:  (400, 400, 426, 428)
'la' anchor bbox:  (400, 411, 426, 439)
Default anchor bbox:  (400, 411, 426, 439)
```

**Comment by djhoese** [comment]
For what it's worth, my workaround for my problem of getting the overall offset of the text height was to take `bottom - pos[1]` which then includes the extra space added on by the ascender anchor.

**Comment by nulano** [comment]
I suppose it's not perfectly obvious, but "top left" is an ambiguous term; both `la` and `lt` are "top left", for different definitions of "top". This is clarified at the top of https://pillow.readthedocs.io/en/stable/handbook/text-anchors.html#text-anchors:
> The anchor parameter determines the alignment of drawn text relative to the xy parameter. The default alignment is top left, specifically `la` (left-ascender) for horizontal text and `lt` (left-top) for vertical text.

This was already the default before `anchor`s were implemented.

---

> I expected similar results between the old textsize and the newer textbbox so I could adjust the position of the text for the old code I was updating.

The default anchor for both is the same, however `textsize` had a bug¹ where it didn't include the top offset. This is explained in the documentation for the (now removed) `textsize` function:
https://github.com/python-pillow/Pillow/blob/204590600c6a8c5246d0564d3312abd6e6b0ed70/docs/reference/ImageDraw.rst?plain=1#L489-L492

In other words, the `textsize` function returned the following tuple:
```python
left, top, right, bottom = font.getbbox(text)
assert font.getsize(text) == (right - left, bottom)
```

¹ *It is unclear whether it is a documentation bug or an implementation bug, because the word "height" is ambiguous.*

**Comment by djhoese** [comment]
If this is clear to others then no need to keep this open. I'll close this since my questions were answered.

**Comment by djhoese** [comment]
> The text anchor alignment. Determines the relative location of the anchor to the text. The default alignment is top left, either la or lt. See [Text anchors](https://pillow.readthedocs.io/en/stable/handbook/text-anchors.html#text-anchors) for details. This parameter is ignored for non-TrueType fonts.

Thoughts on "either la (horizontal) or lt (vertical)"? Or some other indication about which default is used when?

**Comment by nulano** [comment]
That doesn't mention non-TrueType fonts, but I think this would be good:

> The text anchor alignment. Determines the relative location of the anchor to the text. The default alignment is top left, either `la` (horizontal text with TrueType font) or `lt` (otherwise). See [Text anchors](https://pillow.readthedocs.io/en/stable/handbook/text-anchors.html#text-anchors) for details. This parameter is ignored for non-TrueType fonts.

**Comment by djhoese** [comment]
Ah I see. Good point on the "otherwise". Given the length of the parenthetical, you could just make it part of the sentence:

> The default alignment is top left which is "la" for horizontal text with TrueType fonts or "lt" otherwise.

Not sure I understood this or if I just missed it, but besides historical and backwards compatibility reasons, shouldn't the default alignment be "lt" for all orientations and fonts? Just for consistency?

**Comment by djhoese** [comment]
Ok so the "la" horizontal anchor is ideal but only available for some fonts. So the choice was best possible option versus consistent behavior (same anchor) of inconsistent placement (dependent on character height). Makes sense. I just wanted to make sure I was understanding the decision.

**Comment by nulano** [comment]
I've created #7600 to update the documentation.
